spec/unit/actions/collect_data_product_spec.rb

```diff
@@ -0,0 +1,82 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+describe GoodData::LCM2::CollectDataProduct do
+  it 'Has GoodData::Bricks::DefaultDataProductMiddleware class' do
+    GoodData::LCM2::CollectDataProduct.should_not be(nil)
+  end
+
+  let(:client) { double(:client) }
+  let(:domain) { double(:domain) }
+  let(:data_product) { GoodData::DataProduct.new({}) }
+  let(:gdc_logger) { double(:gdc_logger) }
+
+  before do
+    allow(client).to receive(:domain) { domain }
+    allow(gdc_logger).to receive(:info) {}
+  end
+
+  context 'when existing data_product parameter is passed' do
+    before do
+      allow(domain).to receive(:data_products) { data_product }
+    end
+
+    let(:params) do
+      GoodData::LCM2.convert_to_smart_hash({
+        data_product: "data-product-#{SecureRandom.uuid}",
+        gdc_gd_client: client,
+        gdc_logger: gdc_logger
+      })
+    end
+
+    it 'hydrates the data_product object into params and runs the app' do
+      result = subject.class.call(params)
+      expect(result[:params][:data_product]).to be_a GoodData::DataProduct
+    end
+  end
+
+  context 'when not existing data_product parameter is passed' do
+    before do
+      allow(domain).to receive(:data_products) { nil }
+      allow(domain).to receive(:create_data_product) { data_product }
+    end
+
+    let(:params) do
+      GoodData::LCM2.convert_to_smart_hash({
+        data_product: "data-product-#{SecureRandom.uuid}",
```

Comment:
Style/IndentHash: Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.